lib/crystalball/graphql/map_generator/graphql_strategy.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+require 'graphql'
+require 'crystalball/map_generator/base_strategy'
+
+module Crystalball
+  module GraphQL
+    module MapGenerator
+      # Map generator strategy to build map of GraphQL type definitions
+      # that might affect an example.
+      # Patches `GraphQL::ObjectType.define` to get the path of files that call it.
+      class GraphQLStrategy
+        include ::Crystalball::MapGenerator::BaseStrategy
+        include ::Crystalball::MapGenerator::Helpers::PathFilter
+
+        class << self
+          attr_accessor :executed
+        end
+
+        attr_accessor :type_definition_paths, :root
+
+        # Tracer to run on every GraphQL query.
+        # Simply sets the executed flag on each execution
+        class Tracer
+          def self.trace(*)
+            GraphQLStrategy.executed = true
+            yield
+          end
+        end
+
+        # Adds the `GraphQLStrategy::Tracer` to all `GraphQL::Schema`s
+        module Patch
+          def initialize
+            super
+            @tracers = [Tracer]
```

Comment:
I dont see where it is used

Reply:
It's internal to `GraphQL::Schema`. http://graphql-ruby.org/queries/tracing.html